eco_parser/element_parsers.py

```diff
@@ -72,10 +78,11 @@ def parse_body(self):
         table_format = self.get_table_format(tbody)
         if table_format == self.FORMAT_ONE_ROW_PARA:
             return self.parse_one_row_table(tbody)
-        elif table_format == self.FORMAT_STANDARD_TABLE:
+        if table_format == self.FORMAT_STANDARD_TABLE:
             return self.parse_standard_table(tbody)
-        elif table_format == self.FORMAT_UNKNOWN:
+        if table_format == self.FORMAT_UNKNOWN:
             raise ParseError("Could not detect table format", 0)
+        return None
```

Comment:
Strictly speaking, this does replicate exactly what the code did before.
I actually think it would be better to just make this method

```py
def parse_body(self):
    tbody = get_single_element(self.element, "html:tbody")
    table_format = self.get_table_format(tbody)
    if table_format == self.FORMAT_ONE_ROW_PARA:
        return self.parse_one_row_table(tbody)
    if table_format == self.FORMAT_STANDARD_TABLE:
        return self.parse_standard_table(tbody)
    raise ParseError("Could not detect table format", 0)
```

There's no code that handles this returning None in any sensible way. We expect either a parsed body or an exception.